packages/@aws-cdk/aws-stepfunctions-tasks/lib/codebuild/start-build.ts

```diff
@@ -109,4 +183,17 @@ export class CodeBuildStartBuild extends sfn.TaskStateBase {
       Value: environmentVariables[name].value,
     }));
   }
+
+  private validateOverridingParameters(props: CodeBuildStartBuildProps) {
+    if (props.overrides?.secondaryArtifacts && props.overrides!.secondaryArtifacts!.length > 12) {
```

Comment:
why not just: 

```suggestion
    if (props.overrides?.secondaryArtifacts?.length > 12) {
```

Reply:
Got compiling error `object possible is undefined`